docs/user-guide/creating-application/overview.md

```diff
@@ -47,4 +47,57 @@ Click **Save**. The application will be moved to the selected project.
 * To remove the tags from propagation, click the symbol <img src="https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/propagate-dark.jpg" height="10"> again.
 * Click `Save`.
 
-The changes in the tags will be reflected in the `Tags` on the `Overview` section.
\ No newline at end of file
+The changes in the tags will be reflected in the `Tags` on the `Overview` section.
+
+## Configure PersistentVolumeClaim (PVC) for Build Time Optimization
+
+ A PersistentVolumeClaim (PVC) volume is a request for storage, which is used to mount a PersistentVolume (PV) into a Pod. In order to optimize build time, you can configure PVC in your application.
+
+If you want to optimize build time for the multiple target platforms (e.g., arm64, amd64), mounting PVC will provide volume directly to a pod which helps in shorter build time by storing build cache. Mounting PVC will provide storage for build cache which will not impact the normal build where the image is built on the basis of architecture and operating system of the K8s node on which CI is running.
+
+### Create PVC file
+
+* The following configuration file describes persistent volume claim e.g.,`cache-pvc.yaml`, where you have to define the metadata `name` and `storageClassname`.
+
+```bash
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: cache-pvc{here comes the name of PVC}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: {here comes storage class name}
+  resources:
+    requests:
+      storage: 30Gi
+```
+
+* Create the PersistentVolumeClaim by running the following command:
+
+```bash
+kubectl apply -f https://k8s.io/examples/pods/storage/pv-claim.yaml -n {namespace}
+```
+
+For more detail, refer [Kubernetes PVC](https://kubernetes.io/docs/tasks/configure-pod-container/configure-persistent-volume-storage/#create-a-persistentvolumeclaim).
+
+
+## Configure PVC
+
+In order to configure PVC:
+* Go to the `Overview` section of your application.
+* On the right-corner, click `Edit Tags`.
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/overview/pvc-edit-tags.jpg)
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/creating-application/overview/manage-tags-pvc.jpg)
+
+* For app level PVC mounting, enter the following:<ul><li>key:`devtron.ai/ci-pvc-all`</li><li>value: metadata name (e.g., `cache-pvc)` which you define on the [PVC template](#create-pvc-file).</li></ul>`Note`: This PVC mounting will impact all the build pipilines of the application.
+* For pipeline level, enter the following:<ul><li>key:`devtron.ai/ci-pvc-{pipelinename}`</li><li>value: metadata name which you define on the [PVC template](#create-pvc-file).</li></ul>`Note`: This PVC mounting will impact only the particular build pipeline.
```

Comment:
Also, can we attach a screenshot or describe how user can get pipeline name or which one is the pipeline name user has to use in pvc selector?

Reply:
Yes, we can include that info.

Reply:
Please include that too, as it will help the users to easily find and add pipeline names

Reply:
Included.

Reply:
Looks good